xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/java/org/xwiki/model/validation/edit/EditConfirmationScriptService.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.model.validation.edit;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.rendering.block.GroupBlock;
+import org.xwiki.script.service.ScriptService;
+import org.xwiki.stability.Unstable;
+import org.xwiki.text.XWikiToStringBuilder;
+
+import static org.apache.commons.lang3.exception.ExceptionUtils.getRootCauseMessage;
+
+/**
+ * This class provides the script services for handling document edit confirmation.
+ *
+ * @version $Id$
+ * @since 15.8RC1
+ */
+@Component
+@Singleton
+@Named("editConfirmation")
+@Unstable
+public class EditConfirmationScriptService implements ScriptService
+{
+    @Inject
+    @Named("context")
+    private ComponentManager componentManager;
+
+    @Inject
+    private Logger logger;
+
+    /**
+     * Performs a check by invoking the check method of all available {@link EditConfirmationChecker} components and
+     * aggregating their results.
+     *
+     * @param editForced a boolean value indicating whether the check is forced
+     * @return a {@link CheckResults} object containing the results of the check
+     */
+    public CheckResults check(Boolean editForced)
+    {
+        try {
+            CheckResults result = new CheckResults();
+            boolean unboxedEditForced = Optional.ofNullable(editForced).orElse(false);
+            this.componentManager.<EditConfirmationChecker>getInstanceList(EditConfirmationChecker.class)
```

Comment:
Feels like you might want to document that those components should use priorities so that there's a controlled order of the displayed messages.

Reply:
Done + defined priority on existing components.